qiskit/providers/__init__.py

```diff
@@ -110,6 +109,394 @@
+            return MyJob(self. job_handle, job_json, circuit)
+
+
+Transpiler Interface
+--------------------
+
+The key piece of the backend object is how it describes itself to the compiler.
+This is composed of three classes, the
+:class:`~qiskit.providers.models.BackendConfiguration` which describes the
+immutable properties of the backend (things like the number of qubits, coupling
+map, etc), the :class:`~qiskit.providers.models.BackendProperties` which
+describes measured properties of the device (such as current error rates), and
+the :class:`~qiskit.providers.models.PulseDefaults` which defines the
+default pulse behavior for the backend (such as the pulse sequences for gates).
+Of these the only required piece for a backend is the
+:class:`~qiskit.providers.models.BackendConfiguration`, the other information
+is only used if the backend either returns the necessary information.
+
+It's worth noting that this interface will likely evolve over time as the
+format, details, API, and information of the data the compiler needs from a
+backend will change as the transpiler and backends grow and improve. This is
+a key reason why the providers interfaces are versioned to enable making these
+changes in a controlled manner. The details here will change and reflect the
+latest version of this interface.
+
+Custom Basis Gates
+^^^^^^^^^^^^^^^^^^
+
+If your backend doesn't use gates in the Qiskit circuit library
+(:mod:`qiskit.circuit.library`) you can integrate support for this into your
+provider. The basic method for doing this is first to define a
+:class:`~qiskit.circuit.Gate` class for all the custom gates in the basis
+set. For example::
+
+    import numpy as np
+
+    from qiskit.circuit import Gate
+    from qiskit.circuit import QuantumCircuit
+
+    class SYGate(Gate):
+        def __init__(self, label=None):
+            super().__init__("sy", 1, [], label=label)
+
+        def _define(self):
+            qc = QuantumCircuit(1)
+            q.ry(np.pi / 2, 0)
+            self.definition = qc
+
+The key thing to ensure that any custom gates in your Backend's basis set
+is that your custom gate's name attribute (the first param on
+``super().__init__()`` in the ``__init__`` defintion above does not conflict
+with the name of any other gates. The name attribute is what is used to
+identify the gate in the basis set for the transpiler and if there is a conflict
+the transpiler will not know which gate to use.
+
+After you've defined the custom gates to use for the Backend's basis set
+then you need to add equivalence rules to the standard equivalence library
+so that the :class:`~qiskit.transpiler.passes.BasisTranslator` used in the
+:func:`~qiskit.compiler.transpile` function and :mod:`~qiskit.transpiler`
+module can convert an arbitrary circuit using the custom basis set. This
+can be done by defining equivalent circuits, in terms of the custom gate,
+for standard gates. Typically if you can convert from a
```

Comment:
if you can covert from these, then what?

Reply:
This continues on the next line the full sentence reads   "Typically if you can convert from a `CXGate` (if your basis doesn't include a standard 2 qubit gate) and some commonly used single qubit rotation gates like `HGate`."

There is another issue here though, the next sentence also starts with "Typically", which I'll fix now

Reply:
Oh, nm I see what you're saying there's no finish there. I'll fix that now

Reply:
Fixed in: https://github.com/Qiskit/qiskit-terra/pull/5406/commits/922ae3cc737a960f608e297abdcda8d16f0fb0b0